src/Manifest.ts

```diff
@@ -1,135 +1,284 @@
+  public readonly resources: Array<Link>;
+  public readonly toc: Array<Link>;
+  private readonly manifestUrl: URL;
 
-        // Respond immediately with the manifest from the store, if possible.
-        if (store) {
-            const manifestString = await store.get("manifest");
-            if (manifestString) {
-                // Kick off a fetch to update the store for next time,
-                // but don't await it.
-                tryToUpdateManifestButIgnoreResult();
-                const manifestJSON = JSON.parse(manifestString);
-                return new Manifest(manifestJSON, manifestUrl);
-            }
-        }
+  public static async getManifest(
+    manifestUrl: URL,
+    store?: Store
+  ): Promise<Manifest> {
+    const fetchManifest = async (): Promise<Manifest> => {
+      const isJSONManifest = Boolean(manifestUrl.href.endsWith(".json"));
+
+      const manifestJSON = isJSONManifest
+        ? await window
+            .fetch(manifestUrl.href)
+            .then((response) => response.json())
+        : await window
+            .fetch(manifestUrl.href)
+            .then((response) => response.text())
+            .then((str) =>
+              new window.DOMParser().parseFromString(str, "text/xml")
+            )
+            .then((data) => JSON.stringify(xmlToJson(data)));
+
+      if (store) {
+        await store.set("manifest", JSON.stringify(manifestJSON));
+      }
+      return new Manifest(
+        JSON.parse(JSON.stringify(manifestJSON)),
+        manifestUrl
+      );
+    };
 
-        return fetchManifest();
+    const tryToUpdateManifestButIgnoreResult = async (): Promise<void> => {
+      try {
+        await fetchManifest();
+      } catch (err) {
+        // Ignore errors.
+      }
+      return new Promise<void>((resolve) => resolve());
+    };
+
+    // Respond immediately with the manifest from the store, if possible.
+    if (store) {
+      const manifestString = await store.get("manifest");
+      if (manifestString) {
+        // Kick off a fetch to update the store for next time,
+        // but don't await it.
+        tryToUpdateManifestButIgnoreResult();
+        const manifestJSON = JSON.parse(manifestString);
+        return new Manifest(manifestJSON, manifestUrl);
+      }
     }
 
-    public constructor(manifestJSON: any, manifestUrl: URL) {
-        this.metadata = manifestJSON.metadata || {};
-        this.links = manifestJSON.links || [];
-        this.spine = (manifestJSON.readingOrder || manifestJSON.spine) || [];
-        this.resources = manifestJSON.resources || [];
-        this.toc = manifestJSON.toc || [];
-        this.manifestUrl = manifestUrl;
+    return fetchManifest();
+  }
+
+  parseOPFPackage(OPFPackage: any): any {
+    let OPF = null;
+    try {
+      OPF = JSON.parse(OPFPackage);
+    } catch (e) {
+      OPF = JSON.parse(JSON.stringify(OPFPackage));
     }
+    return OPF;
+  }
+
+  public parseMetaData(OPFPackage: any): any {
```

Comment:
Can we have a block comment making clear that there's a whole block of OPF-specific functions?